### PR TITLE
Add chef generated files and fix up errors encountered

### PR DESCRIPTION
**Pull request by kierenevans**
- Update Gem dependencies, add Berkshelf which then in turn requires ruby version >= 1.9.3 as discovered in https://github.com/inviqa/chef-varnish/pull/20 .
- Fix up issues encountered as a result of a newer foodcritic version.
- Fix up issues encountered when running `kitchen test centos` where the template uses node['solr'], which doesn't exist without depending upon it.

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+driver:
+  name: vagrant
+
+provisioner:
+  name: chef_zero
+
+platforms:
+  - name: ubuntu-12.04
```
Comment (shanethehat) [inline .kitchen.yml]: 12.04 is end of life, do we really need to support it?

File: Berksfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+source "https://supermarket.getchef.com"
+
+metadata
+
+# Solr
+cookbook 'solr',
+    :git => 'git://github.com/inviqa/chef-solr.git',
+    :ref => 'master'
```
Comment (shanethehat) [inline Berksfile]: I don't think we should track the master branch like this.

Comment (andytson) [inline Berksfile]: also `source "https://supermarket.getchef.com"` now `source "https://supermarket.chef.io"` (tho it redirects)

File: .kitchen.yml (hunk 1 above)
Comment (andytson) [inline .kitchen.yml:9]: nope :) if the OS isn't getting security updates, we shouldn't support

File: .kitchen.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  - name: ubuntu-14.04
+  - name: centos-6.5
```
Comment (andytson) [inline .kitchen.yml:11]: also this should be 6.7 (atlas version preferred)

File: .travis.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 language: ruby
 rvm:
-  - 1.9.2
   - 1.9.3
+  - 2.1.5
```
Comment (andytson) [inline .travis.yml:4]: could this be:
- 1.9
- 2.1
- 2.2

It then follows the latest bugfix version

File: .kitchen.yml (hunk 3 above)
Comment (kierenevans) [inline .kitchen.yml:11]: I'll be breaking this PR up into more managable chunks, so I'll address test-kitchen with the latest boxes in a new PR.

**Pull request closed without merging**